wallet/types/src/keys.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crypto::key::extended::{ExtendedPrivateKey, ExtendedPublicKey};
+use serialization::{Decode, Encode};
+use storage::HasPrefix;
+
+/// The key id prefix is used for searching specific kinds of keys
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub enum KeyIdPrefix {
+    #[codec(index = 0)]
+    DeterministicRoot,
+}
+
+/// The key id is described by it's public key
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub enum KeyId {
+    #[codec(index = 0)]
+    DeterministicRoot(ExtendedPublicKey),
+}
+
+impl HasPrefix<KeyIdPrefix> for KeyId {}
+
+/// The useful content of this key e.g. a private key or an address depending on the usage
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub enum KeyContent {
+    #[codec(index = 0)]
+    DeterministicRoot(ExtendedPrivateKey),
+}
```

Comment:
How do you feel about breaking this down into a separate key-value map per key "type"? Is it just a matter of extra boilerplate or is there another reason going with enums is useful here?